### PR TITLE
Bump gl-native to 10.0.0-rc.8, common to 18.0.0

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Bump gl-native to 10.0.0-rc.8, common to 18.0.0</changelog>`.

## Breaking changes ⚠️
* In offline mode (set by either `mapbox::common::OfflineSwitch` API or on platform side), the error notifications are send if the required resources are not present locally. The volatile tiles are not considered to be required in offline. 
* Adapt setBounds to gl-js behavior: constraining of coordinates and zoom level is now stricter to prevent out of bounds map area to be visible in the viewport 
* Add HTTP interceptor API - for anyone who is using HttpServiceInterface; there is a new method called setInterceptor that should be overridden
## Features ✨ and improvements 🏁
* Improve collision detection by using runtime calculated sizes for collision boxes. Previously collision boxes' sizes are constant, they are calculated during symbol layout time by using constant zoom level 
* Improve collision detection by using runtime calculated pixelated sizes for collision circles. Previously collision circles' sizes are constant, they are calculated during symbol layout time by using constant zoom level  
* Implement 'promoteId' feature for geojson and vector sources. The feature allows to promote feature's property to a feature id, so that promoted id can be used with FeatureState API. 
* Enable instant  transitions for data-driven paint layer properties 

## Bug fixes 🐞
* Fix volatile tiles disappearing on "not modified" response 
* Prioritize addition of a persistent layer whose id is used for other persistent layer positions 
* Only do line breaking process for point placement labels. And if text-max-width is 0, still do general ideographic beaking checks for point labels. 
* Fix collision box's 'dynamicVerticesExt' updating in placement stage 
* Trigger map redraw when feature state changes 
* Print nothing when DisableAll set for log level

### CODE REVIEW

**Review by tobrun — APPROVED** [review]
Could you explain why the code changes were required?

**Review by pengdev — APPROVED** [review]
Shall we bring gl-native and common change log together in this PR?

**Comment by Chaoba** [comment]
@tobrun For offline part because gl-native changed the logic to send error notification; for Point init values is because we add value range for latitude and longitude.
@pengdev Add change log to pr description

File: app/src/main/java/com/mapbox/maps/testapp/examples/customlayer/CustomLayerActivity.kt [hunk 1]
```diff
@@ -29,7 +29,7 @@ class CustomLayerActivity : AppCompatActivity() {
       Style.MAPBOX_STREETS
     ) {
       mapboxMap.setCamera(
-        CameraOptions.Builder().center(Point.fromLngLat(-243.60947, 39.91448)).zoom(10.0).build()
+        CameraOptions.Builder().center(Point.fromLngLat(-63.60947, 39.91448)).zoom(10.0).build()
```
Comment (kiryldz) [inline app/src/main/java/com/mapbox/maps/testapp/examples/customlayer/CustomLayerActivity.kt:32]: It should be just `63` I assume

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt [hunk 2]
```diff
@@ -419,9 +422,11 @@ class OfflineTest {
     }
     switchAirplaneMode()
     verifyAirplaneModeEnabled()
+    // Show map with the zoom level larger than download range to trigger loading error
     prepareMapView(
       observer,
-      listOf(MapEvents.MAP_IDLE, MapEvents.MAP_LOADING_ERROR)
+      listOf(MapEvents.MAP_IDLE, MapEvents.MAP_LOADING_ERROR),
+      ZOOM + 3
```
Comment (kiryldz) [inline sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:429]: Why is it 3 here? 1 in some places and 2 in some other places.
Perhaps we could use one number and move it to const.

Comment (Chaoba) [reply sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:429]: The download rang is [zoom - 2, zoom + 2], use zoom + 3 here to trigger loading error.

File: app/src/main/java/com/mapbox/maps/testapp/examples/customlayer/CustomLayerActivity.kt (hunk 1 above)
Comment (Chaoba) [reply app/src/main/java/com/mapbox/maps/testapp/examples/customlayer/CustomLayerActivity.kt:32]: Should be 116.39053

**Review by kiryldz — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Pull request merged**